### PR TITLE
#17363 : Passing ProfilerChan by reference to the send_profile_data method.

### LINKED ISSUE #17363: Make send_profile_data take profiler_chan argument by reference
This will allow us to pass borrowed channel arguments to it, rather than unnecessarily cloning them.

Code: `components/profile_traits/time.rs`

### PR DESCRIPTION
**Pull request by o0Ignition0o**
Passed the ProfilerChan by reference to the send_profile_data method. 

I wonder if I should also refactor the 
`pub fn profile<T, F>(category: ProfilerCategory, meta: Option<TimerMetadata>, profiler_chan: ProfilerChan, callback: F) -> T where F: FnOnce() -> T method ` , but I don't feel confident enough to make the call, since I don't really understand what it would imply.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #17363 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because the project would not compile if there were missing by reference calls.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17383)

### CODE REVIEW

**Comment by jdm** [comment]
I would go ahead and make the change to `profile` as well. That seems like a worthwhile change to me!

File: components/compositing/compositor.rs [hunk 1]
```diff
@@ -1483,7 +1483,7 @@ impl<Window: WindowMethods> IOCompositor<Window> {
             _ => initialize_png(&*self.gl, width, height)
         };
 
-        profile(ProfilerCategory::Compositing, None, self.time_profiler_chan.clone(), || {
+        profile(ProfilerCategory::Compositing, None, &self.time_profiler_chan.clone(), || {
```
Comment (jdm) [inline components/compositing/compositor.rs]: All of the cases that use `&foo.clone()` should be able to use `&foo` instead.

**Review by jdm — CHANGES_REQUESTED** [review]

File: components/compositing/compositor.rs (hunk 1 above)
Comment (o0Ignition0o) [inline components/compositing/compositor.rs]: Hey @jdm, and thanks for the review ! Could you point me to any hint which would allow me to pass &self.time_profiler_chan to the closure without cloning it first ? The compiler throws some borrow issues at me : 

` error[E0500]: closure requires unique access to `self` but `self.time_profiler_chan` is already borrowed
    --> /home/ignition/Documents/Projects/rust/servo/components/compositing/compositor.rs:1486:80
     |
1486 |         profile(ProfilerCategory::Compositing, None, &self.time_profiler_chan, || {
     |                                                       -----------------------  ^^ closure construction occurs here
     |                                                       |
     |                                                       borrow occurs here
...
1490 |             let size = webrender_traits::DeviceUintSize::from_untyped(&self.frame_size.to_untyped());
     |                                                                        ---- borrow occurs due to use of `self` in closure
1491 |             self.webrender.render(size);
1492 |         });
     |          - borrow ends here

error: aborting due to previous error(s)
 `

Comment (jdm) [inline components/compositing/compositor.rs]: Ah, I see. In that case, let's revert the change to profile - I suspect many people will make the same assumption that I did (that & followed by clone() is unnecessary) and run into this same error.

Comment (o0Ignition0o) [inline components/compositing/compositor.rs]: Hmm ok then, I was considering changing the closures, or working with the move keyword, but the first option would be less readable, and the second one would be neither correct nor performant. I'll revert my changes, I wish I could find an elegant way to refactor it but I can't find any :/

**Comment by o0Ignition0o** [comment]
I've reverted the profile function refactorization, I guess it should be it

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!
